### PR TITLE
net-voip/sipvicious: version 0.3.4

### PR DESCRIPTION
**Pull request by jkroonza**
Package-Manager: Portage-3.0.20, Repoman-3.0.2
Signed-off-by: Jaco Kroon <jaco@uls.co.za>

### CODE REVIEW

File: net-voip/sipvicious/sipvicious-0.3.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2020-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7..10} )
```
Comment (ionenwks) [inline net-voip/sipvicious/sipvicious-0.3.4.ebuild:6]: 7 is dead code (not recognized by eclasses anymore), good to cleanup at same time if you're going to change this line.

File: net-voip/sipvicious/sipvicious-0.3.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2020-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (ionenwks) [inline net-voip/sipvicious/sipvicious-0.3.4.ebuild:4]: maybe time for EAPI-8

File: net-voip/sipvicious/sipvicious-0.3.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+PYTHON_COMPAT=( python3_{7..10} )
+DISTUTILS_USE_SETUPTOOLS=rdepend
+DISTUTILS_SINGLE_IMPL=1
+
+inherit distutils-r1
+
+DESCRIPTION="SIPVicious is a SIP security package"
+HOMEPAGE="https://github.com/EnableSecurity/sipvicious/wiki"
+SRC_URI="https://github.com/EnableSecurity/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+KEYWORDS="~amd64 ~x86"
```
Comment (ionenwks) [inline net-voip/sipvicious/sipvicious-0.3.4.ebuild:15]: Just style but, to match skel.ebuild ordering, KEYWORDS normally would go straight below SLOT

File: net-voip/sipvicious/sipvicious-0.3.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+LICENSE="GPL-3+"
+SLOT="0"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
```
Comment (ionenwks) [inline net-voip/sipvicious/sipvicious-0.3.4.ebuild:20]: With distutils-r1 you don't need to define this, only needed with python-* eclasses (or if you set DISTUTILS_OPTIONAL, but it's not here)

File: net-voip/sipvicious/sipvicious-0.3.4.ebuild [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+RDEPEND="$(python_gen_cond_dep '
+		dev-python/dnspython[${PYTHON_USEDEP}]
+		dev-python/reportlab[${PYTHON_USEDEP}]
+	')"
+
+src_prepare() {
+	default
+	sed -re 's:man/man1:share/man/man1:' -i setup.py || die "Error updating man page installation location."
```
Comment (ionenwks) [inline net-voip/sipvicious/sipvicious-0.3.4.ebuild:29]: Optional but personally think an elaborate `die` message is not needed for a self-explanatory sed (the line will be displayed on error), so just `|| die` is fine.

This also doesn't use extended regex, so -r is superfluous.

File: net-voip/sipvicious/sipvicious-0.3.4.ebuild [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2020-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7..10} )
+DISTUTILS_USE_SETUPTOOLS=rdepend
```
Comment (ionenwks) [inline net-voip/sipvicious/sipvicious-0.3.4.ebuild:7]: Can remove this
```
 * QA Notice: DISTUTILS_USE_SETUPTOOLS value is probably incorrect
 *   have:     DISTUTILS_USE_SETUPTOOLS=rdepend
 *   expected: (unset)
```

File: net-voip/sipvicious/sipvicious-0.3.4.ebuild [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+DISTUTILS_SINGLE_IMPL=1
+
+inherit distutils-r1
+
+DESCRIPTION="SIPVicious is a SIP security package"
+HOMEPAGE="https://github.com/EnableSecurity/sipvicious/wiki"
+SRC_URI="https://github.com/EnableSecurity/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (ionenwks) [inline net-voip/sipvicious/sipvicious-0.3.4.ebuild:14]: Could plain-text this ${PN}, see:
https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide/Style_Guide#Plaintexting_.24.7BPN.7D_in_SRC_URI

File: net-voip/sipvicious/sipvicious-0.3.4.ebuild (hunk 6 above)
Comment (ionenwks) [reply net-voip/sipvicious/sipvicious-0.3.4.ebuild:7]: Any update on addressing some of the other points I mentioned like this one? (there are others I haven't heard back from)

There's even a bug opened for this now
https://bugs.gentoo.org/809473

File: net-voip/sipvicious/sipvicious-0.3.4.ebuild (hunk 5 above)
Comment (jkroonza) [reply net-voip/sipvicious/sipvicious-0.3.4.ebuild:29]: I still prefer keeping the text, what you consider self-explanatory may not be for others.

File: net-voip/sipvicious/sipvicious-0.3.4.ebuild (hunk 7 above)
Comment (jkroonza) [reply net-voip/sipvicious/sipvicious-0.3.4.ebuild:14]: And plaintexting this makes additional requirement for change on version to version ... so how do you balance these conflicting ideals?  emerge -pf is useful and easy for the case described given in the referenced link, even if not plaintexted here.

Comment (ionenwks) [reply net-voip/sipvicious/sipvicious-0.3.4.ebuild:13]: This was only about ${PN}, when does ${PN} change?

**Review by ionenwks — APPROVED** [review]

**Pull request closed without merging**